server/index.js

```diff
@@ -10,23 +11,42 @@ app.listen(`${port}`, () => {
 });
 
 app.get('/products/:id', (req, res, next) => {
-  let data = [];
-  api.get("products", req.params.id, (err, result) => {
+  const data = [];
+  api.get('products', req.params.id, (err, result) => {
     if (err) {
       console.log('There was an error getting products from database: ', err);
       next();
     } else {
-        data.push(result);
-        let shopId = JSON.parse(result)[0].shop_id;
-        api.get("shops", shopId, (shopErr, shopResult) => {
+      data.push(result);
+      const shopId = JSON.parse(result)[0].shop_id;
+      api.get('shops', shopId, (shopErr, shopResult) => {
```

Comment:
It looks like you wind up parsing it here, but if you didnt stringify it in your api then would this even be necessary?